rosidl_typesupport_c/src/type_support_dispatch.hpp

```diff
@@ -44,44 +43,49 @@ get_typesupport_handle_function(
     return handle;
   }
 
-#ifdef ROSIDL_TYPESUPPORT_C_USE_POCO
   if (handle->typesupport_identifier == rosidl_typesupport_c__typesupport_identifier) {
     const type_support_map_t * map = \
       static_cast<const type_support_map_t *>(handle->data);
     for (size_t i = 0; i < map->size; ++i) {
       if (strcmp(map->typesupport_identifier[i], identifier) != 0) {
         continue;
       }
-      Poco::SharedLibrary * lib = nullptr;
+      rcpputils::SharedLibrary * lib = nullptr;
+
       if (!map->data[i]) {
         char library_name[1024];
         snprintf(
           library_name, 1023, "%s__%s",
           map->package_name, identifier);
         std::string library_path = rcpputils::find_library_path(library_name);
         if (library_path.empty()) {
-          fprintf(stderr, "Failed to find library '%s'\n", library_name);
-          return nullptr;
+          throw std::runtime_error("Failed to find library '" + std::string(library_name) + "'");
+        }
+
+        try {
+          lib = new rcpputils::SharedLibrary(library_path.c_str());
+        } catch (const std::runtime_error & e) {
+          throw std::runtime_error("Could not load library " + std::string(e.what()));
+        } catch (const std::bad_alloc & e) {
+          throw std::bad_alloc();
         }
-        lib = new Poco::SharedLibrary(library_path);
         map->data[i] = lib;
       }
-      auto clib = static_cast<const Poco::SharedLibrary *>(map->data[i]);
-      lib = const_cast<Poco::SharedLibrary *>(clib);
-      if (!lib->hasSymbol(map->symbol_name[i])) {
-        fprintf(stderr, "Failed to find symbol '%s' in library\n", map->symbol_name[i]);
-        return nullptr;
+      auto clib = static_cast<const rcpputils::SharedLibrary *>(map->data[i]);
+      lib = const_cast<rcpputils::SharedLibrary *>(clib);
+      if (!lib->has_symbol(map->symbol_name[i])) {
+        throw std::runtime_error(
+                "Failed to find symbol '" + std::string(
+                  map->symbol_name[i]) + "' in library");
```

Comment:
I think you should prefer to split lines at the + operations rather than in the string call.